src/groupByKey.ts

```diff
@@ -1,7 +1,18 @@
-type GroupsMap<T> = {
-  [key: string]: T[];
-};
-
-export function groupByKey(items, key) {
+export function groupByKey<T>(items: T[], key: keyof T): object {
   // write code here;
+  const result: any = {};
+
+  items.forEach((itemsElement: T) => {
+    const variable: string = String(itemsElement[key]);
+
+    if (Object.prototype.hasOwnProperty.call(result, variable)) {
```

Comment:
```suggestion
    if (result.hasOwnProperty(variable)) {
```

Reply:
If I use code like this, I got error:
  12:16  error  Do not access Object.prototype method 'hasOwnProperty' from target object  no-prototype-builtins